salt/_modules/kubernetes.py

```diff
@@ -1975,3 +1992,162 @@ def replace_daemonset(
             raise CommandExecutionError(exc)
     finally:
         _cleanup(**cfg)
+
+
+def show_clusterrole(name, **kwargs):
+    cfg = _setup_conn(**kwargs)
+    try:
+        api_instance = kubernetes.client.RbacAuthorizationV1Api()
+        api_response = api_instance.read_cluster_role(name)
+
+        return api_response.to_dict()
+    except (ApiException, HTTPError) as exc:
+        if isinstance(exc, ApiException) and exc.status == 404:
+            return None
+        else:
+            log.exception(
+                'Exception when calling '
+                'RbacAuthorizationV1Api->read_cluster_role'
+            )
+            raise CommandExecutionError(exc)
+    finally:
+        _cleanup(**cfg)
```

Comment:
You can definitely mutualize this whole try/except/finally pattern between the different methods

Reply:
Yes it's something we need to do.
Btw this file and the one in `_states` directory it's just a copy from the `salt` repo with our new function, which is, for the moment, copy/paste for new kubernetes stuff (maybe in futur version of salt we will remove this file + the `_states/kubernete.py`).
So the structure is the same as others salt function.